hypershift-operator/controllers/awsendpointservice/awsendpointservice_controller.go

```
@@ -0,0 +1,208 @@
+			return err
+		}
+		if len(output.ServiceConfigurations) == 0 {
+			// clear the EndpointServiceName so a new Endpoint Service is created on the requeue
+			awsEndpointService.Status.EndpointServiceName = ""
+			return fmt.Errorf("endpoint service %s not found", serviceName)
+		}
+		return nil
+	}
+
+	// determine the LB ARN
+	lbName := awsEndpointService.Spec.NetworkLoadBalancerName
+	output, err := elbv2Client.DescribeLoadBalancersWithContext(ctx, &elbv2.DescribeLoadBalancersInput{
+		Names: []*string{aws.String(lbName)},
+	})
+	if err != nil {
+		return err
+	}
+	if len(output.LoadBalancers) == 0 {
+		return fmt.Errorf("NLB %s not found", lbName)
+	}
+	lbARN := output.LoadBalancers[0].LoadBalancerArn
+
+	// create the Endpoint Service
+	createEndpointServiceOutput, err := ec2Client.CreateVpcEndpointServiceConfigurationWithContext(ctx, &ec2.CreateVpcEndpointServiceConfigurationInput{
+		AcceptanceRequired:      aws.Bool(false),
+		NetworkLoadBalancerArns: []*string{lbARN},
+	})
+	if err != nil {
+		return err
+	}
+	serviceName = *createEndpointServiceOutput.ServiceConfiguration.ServiceName
+	log.Info("endpoint service created", "serviceName", serviceName)
+	awsEndpointService.Status.EndpointServiceName = serviceName
+
+	return nil
+}
+
+func (r *AWSEndpointServiceReconciler) delete(ctx context.Context, awsEndpointService *hyperv1.AWSEndpointService) (bool, error) {
+	serviceName := awsEndpointService.Status.EndpointServiceName
+	if len(serviceName) == 0 {
+		// nothing to clean up
+		return true, nil
+	}
+
+	// parse serviceID from serviceName e.g. com.amazonaws.vpce.us-west-1.vpce-svc-014f44db649a87c02 -> vpce-svc-014f44db649a87c02
+	parts := strings.Split(serviceName, ".")
+	serviceID := parts[len(parts)-1]
+
+	// delete the Endpoint Service
+	if _, err := r.ec2Client.DeleteVpcEndpointServiceConfigurationsWithContext(ctx, &ec2.DeleteVpcEndpointServiceConfigurationsInput{
+		ServiceIds: []*string{aws.String(serviceID)},
+	}); err != nil {
+		return false, err
```

Should not-found errors be ignored?

---

Yes, they should be

---

confirmed that there is no error on not found.  just reports successful deletion.